crates/op-rbuilder/src/builders/context.rs

```diff
@@ -399,6 +402,20 @@ impl<ExtraCtx: Debug + Default> OpPayloadBuilderCtx<ExtraCtx> {
                 }
             }
 
+            if let Some(flashblock_index) = flashblock_number {
+                if let Some(flashblock_number_min) = flashblock_number_min {
+                    if flashblock_index < flashblock_number_min {
+                        continue;
```

Comment:
for now a wrapper around the current best_tx iterator to hold the transactions in a hash map and do the flashblocks number filtering would probably be cleaner here